modules/pricer/src/test/java/com/opengamma/strata/pricer/calibration/CalibrationZeroRateUsd2OisFuturesIrsTest.java

```diff
@@ -0,0 +1,393 @@
+  private static final DiscountingSwapProductPricer SWAP_PRICER =
+      DiscountingSwapProductPricer.DEFAULT;
+  private static final DiscountingTermDepositProductPricer DEPO_PRICER =
+      DiscountingTermDepositProductPricer.DEFAULT;
+  private static final MarketQuoteSensitivityCalculator MQC = MarketQuoteSensitivityCalculator.DEFAULT;
+
+  private static final CalibrationMeasures CALIBRATION_MEASURES = CalibrationMeasures.DEFAULT;
+  private static final CurveCalibrator CALIBRATOR = CurveCalibrator.of(1e-9, 1e-9, 100, CALIBRATION_MEASURES);
+
+  // Constants
+  private static final double TOLERANCE_PV = 1.0E-6;
+  private static final double TOLERANCE_PV_DELTA = 1.0E+3;
+
+  private static final CurveGroupName CURVE_GROUP_NAME = CurveGroupName.of("USD-DSCON-LIBOR3M");
+  private static final InterpolatedNodalCurveDefinition DSC_CURVE_DEFN =
+      InterpolatedNodalCurveDefinition.builder()
+          .name(DSCON_CURVE_NAME)
+          .xValueType(ValueType.YEAR_FRACTION)
+          .yValueType(ValueType.ZERO_RATE)
+          .dayCount(CURVE_DC)
+          .interpolator(INTERPOLATOR_LINEAR)
+          .extrapolatorLeft(EXTRAPOLATOR_FLAT)
+          .extrapolatorRight(EXTRAPOLATOR_FLAT)
+          .nodes(DSC_NODES).build();
+  private static final InterpolatedNodalCurveDefinition FWD3_CURVE_DEFN =
+      InterpolatedNodalCurveDefinition.builder()
+          .name(FWD3_CURVE_NAME)
+          .xValueType(ValueType.YEAR_FRACTION)
+          .yValueType(ValueType.ZERO_RATE)
+          .dayCount(CURVE_DC)
+          .interpolator(INTERPOLATOR_LINEAR)
+          .extrapolatorLeft(EXTRAPOLATOR_FLAT)
+          .extrapolatorRight(EXTRAPOLATOR_FLAT)
+          .nodes(FWD3_NODES).build();
+  private static final CurveGroupDefinition CURVE_GROUP_CONFIG =
+      CurveGroupDefinition.builder()
+          .name(CURVE_GROUP_NAME)
+          .addCurve(DSC_CURVE_DEFN, USD, USD_FED_FUND)
+          .addForwardCurve(FWD3_CURVE_DEFN, USD_LIBOR_3M).build();
+
+  //-------------------------------------------------------------------------
+  public void calibration_present_value_oneGroup() {
+    ImmutableRatesProvider result =
+        CALIBRATOR.calibrate(CURVE_GROUP_CONFIG, VALUATION_DATE, ALL_QUOTES, TS, FxMatrix.empty());
+    assertPresentValue(result);
+  }
+  
+  public void calibration_market_quote_sensitivity_one_group() {
+    double shift = 1.0E-6;
+    Function<ImmutableObservableValues, ImmutableRatesProvider> f =
+        (ov) -> CALIBRATOR.calibrate(CURVE_GROUP_CONFIG, VALUATION_DATE, ov, TS, FxMatrix.empty());
+    calibration_market_quote_sensitivity_check(f, shift);
+  }
+
+  private void calibration_market_quote_sensitivity_check(
+      Function<ImmutableObservableValues, ImmutableRatesProvider> calibrator,
+      double shift) {
+    double notional = 100_000_000.0;
+    double spread = 0.0050;
+    SwapTrade trade = FixedIborSwapConventions.USD_FIXED_1Y_LIBOR_3M.toTrade(VALUATION_DATE, Period.ofMonths(8),
```

Comment:
Coding standards would break this line after `toTrade(`. We avoid having a line break in the middle of the parameter list like this.